st2actions/st2actions/workflows/workflows.py

```diff
@@ -40,10 +47,17 @@
     queues.WORKFLOW_ACTION_EXECUTION_UPDATE_QUEUE,
 ]
 
+WORKFLOW_ENGINE = "workflow_engine"
+WORKFLOW_ENGINE_START_STOP_SEQ = "workflow_engine_start_stop_seq"
+
 
 class WorkflowExecutionHandler(consumers.VariableMessageHandler):
     def __init__(self, connection, queues):
         super(WorkflowExecutionHandler, self).__init__(connection, queues)
+        self._active_messages = 0
+        self._semaphore = Semaphore()
+        # This is required to ensure workflows stuck in pausing state after shutdown transition to paused state after engine startup.
+        self._delay = 30
```

Comment:
Answers inline.
1. Do you know what happens if no new WFE comes up? The action execution completion message will stay in rabbitMQ?
-> Right. Since resume flow happens only once during WFE startup, until the message is consumed by a new WFE it'll stay in the queue.

2. What if this is a complete shutdown and the rabbitMQ is rebooted as well?
-> Queues declared in StackStorm are durable and the messages are sent with delivery mode "persistent" (default setting). This means that unless someone purges the data manually, all the messages persist across rabbitMQ shutdown.

3. If the messages in rabbitMQ is lost, how should we handle workflows in PAUSING?
-> The only way a message will get lost here is when someone purges data. Having said that, I think we still need some way to handle workflows in PAUSING. Do you think instead of resuming the workflows paused by shutdown routine only during WFE startup, we could check and resume such workflows at defined intervals?